From: Robin Holt <holt@sgi.com>
Subject: x86: UV - Introduce uv_gpa_is_mmr.
References: bnc#561939, fate#306952
Patch-mainline: 2.6.33-rc1
Git-commit: fae419f2abd15ab7d1cd1413e6683a276a4e14e2

Provide a mechanism for determining if a global physical address is
pointing to a UV hub MMR.

Signed-off-by: Robin Holt <holt@sgi.com>
Signed-off-by: Rafael J. Wysocki <rjw@suse.de>
---

 arch/x86/include/asm/uv/uv_hub.h |    7 +++++++
 1 file changed, 7 insertions(+)
Index: linux-2.6.32-master/arch/x86/include/asm/uv/uv_hub.h
===================================================================
--- linux-2.6.32-master.orig/arch/x86/include/asm/uv/uv_hub.h
+++ linux-2.6.32-master/arch/x86/include/asm/uv/uv_hub.h
@@ -232,6 +232,13 @@ static inline unsigned long uv_gpa(void 
 	return uv_soc_phys_ram_to_gpa(__pa(v));
 }
 
+/* Top two bits indicate the requested address is in MMR space.  */
+static inline int
+uv_gpa_in_mmr_space(unsigned long gpa)
+{
+	return (gpa >> 62) == 0x3UL;
+}
+
 /* UV global physical address --> socket phys RAM */
 static inline unsigned long uv_gpa_to_soc_phys_ram(unsigned long gpa)
 {
